runtime/drivers/duckdb/config.go

```diff
@@ -64,12 +70,26 @@ func newConfig(cfgMap map[string]any) (*config, error) {
 
 		// Remove from query string (so not passed into DuckDB config)
 		qry.Del("rill_pool_size")
-
-		// Rebuild DuckDB DSN (which should be "path?key=val&...")
-		uri.RawQuery = qry.Encode()
-		cfg.DSN = uri.String()
+	}
+	if cfg.MemoryLimitGB > 0 {
+		qry.Add("max_memory", fmt.Sprintf("%dGB", cfg.MemoryLimitGB))
+	}
+	if cfg.CPU > 0 {
+		threads := int(cpuThreadRatio * float64(cfg.CPU))
+		if threads <= 0 {
+			threads = 1
+		}
+		qry.Add("threads", strconv.Itoa(threads))
+		if cfg.CPU <= threads {
+			cfg.PoolSize = cfg.CPU
+		} else {
+			cfg.PoolSize = threads
+		}
```

Comment:
Can we also make a change so the pool size is always at least 2? Having it at 1 is not great with async reconciles. Would need to change line 40, 94, and then here it should be `max(2, min(cfg.CPU, threads))`.